docs/general-development/pagination-for-large-result-sets.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Pagination for large result sets
+ms.date: 10/14/2020
+ms.prod: sharepoint
+localization_priority: Priority
+---
+
+# Pagination for large result sets
+
+If you have a large number of search results (for example, over 50,000) to page through in a query, it is recommended to use the approach explained in this article instead of the approach of [StartRow](https://docs.microsoft.com/sharepoint/dev/general-development/sharepoint-search-rest-api-overview#startrow). This approach uses sorting on `[docid]` in ascending order and uses a query restriction on the `IndexDocId` value with an increasing value of `IndexDocID` for each new page.
+
+The advantages of this approach are:
+
+- It provides pagination with better performance
+- It does not have a limit on the number of pages (if you use the approach of [StartRow](https://docs.microsoft.com/sharepoint/dev/general-development/sharepoint-search-rest-api-overview#startrow) and the `StartRow` value is greater than 50,000, there is a risk of being throttled by SharePoint)
+
+Here is an example of using this approach:
+
+For Page 1, issue a query with sort on `[docid]` in ascending order:
+
+```http
+GET http:// _server_/_api/search/query?querytext='sharepoint'&amp;sortlist='[docid]:ascending'
```

Comment:
Please fix URL reference (_server_). If that's a property that should be replaced use something like {SHAREPOINT_SITE_URL}.